Applications/spire/source/charting/cached_chart_model.cpp

```diff
@@ -51,23 +51,23 @@ QtPromise<std::vector<Candlestick>> CachedChartModel::load_data(
   return m_chart_model->load(data.m_start, data.m_end).then(
     [=] (auto result) {
       auto first = std::lower_bound(m_loaded_data.begin(),
-        m_loaded_data.end(), data.m_start, [=] (const auto& lhs,
+        m_loaded_data.end(), data.m_start, [] (const auto& lhs,
             const auto& rhs) {
           return lhs.GetStart() < rhs;
         });
       auto last = std::lower_bound(m_loaded_data.begin(),
-        m_loaded_data.end(), data.m_end, [=] (const auto& lhs,
+        m_loaded_data.end(), data.m_end, [] (const auto& lhs,
             const auto& rhs) {
           return lhs.GetStart() < rhs;
         });
       auto pos = m_loaded_data.erase(first, last);
       m_loaded_data.insert(pos, result.Get().begin(), result.Get().end());
       auto range_first = std::lower_bound(m_ranges.begin(), m_ranges.end(),
-        data, [=] (const auto& lhs, const auto& rhs) {
+        data, [] (const auto& lhs, const auto& rhs) {
           return lhs.m_start < rhs.m_start;
         });
       auto range_last = std::lower_bound(m_ranges.begin(), m_ranges.end(),
```

Comment:
Your comparators are inconsistent. Either ranges is sorted by ```m_start``` or it's sorted by ```m_end```. You will need to formalize how elements are stored in ```m_ranges``` and apply that ordering consistently.